### PR TITLE
Support placeholders use in repository name for download, move and copy

### PR DESCRIPTION
**Pull request by barbelity**

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -185,24 +185,35 @@ func pathToRegExp(localPath string) string {
...
 //      returns "hello"
 // Example 2:
 //      pattern = "repoA/1(.*)234" ; path = "repoB/1hello234" ; target = "{1}" ; ignoreRepo = true
 //      returns "hello"
 func BuildTargetPath(pattern, path, target string, ignoreRepo bool) (string, error) {
-	if ignoreRepo {
+	asteriskIndex := strings.Index(pattern, "*")
+	slashIndex := strings.Index(pattern, "/")
+	if shouldRemoveRepo(ignoreRepo, asteriskIndex, slashIndex) {
+		// Required when working with virtual repositories as the pattern may contain the virtual-repository name,
```
Comment (eyalbe4) [inline utils/utils.go:202]: Required when ... --> Required the repository part of the path, when ...

File: utils/utils.go [hunk 2]
```diff
@@ -185,24 +185,35 @@ func pathToRegExp(localPath string) string {
...
+		// but the path will contain the local-repository name.
 		pattern = removeRepoFromPath(pattern)
 		path = removeRepoFromPath(path)
 	}
 	pattern = addEscapingParentheses(pattern, target)
 	pattern = pathToRegExp(pattern)
+	if slashIndex < 0 {
+		// '/' doesn't exists, add an optional trailing-slash to support cases in which the provided pattern
```
Comment (eyalbe4) [inline utils/utils.go:210]: '/' doesn't exists... --> If '/' doesn't existt...

File: utils/utils.go (hunk 1 above)
Comment (barbelity) [reply utils/utils.go:202]: What do you think of the following:
`Removing the repository part of the path is required when working with virtual repositories, as the pattern may contain the virtual-repository name, but the path contains the local-repository name.`

File: utils/utils.go (hunk 2 above)
Comment (barbelity) [reply utils/utils.go:210]: Fixed.

File: utils/utils.go (hunk 1 above)
Comment (eyalbe4) [reply utils/utils.go:202]: LGTM.

**Pull request merged**